test/test_stat.rs

```diff
@@ -102,3 +103,51 @@ fn test_stat_fstat_lstat() {
     let fstat_result = fstat(link.as_raw_fd());
     assert_stat_results(fstat_result);
 }
+
+#[test]
+fn test_fchmod() {
+    let tempdir = TempDir::new("nix-test_fchmod").unwrap();
+    let filename = tempdir.path().join("foo.txt");
+    let file = File::create(&filename).unwrap();
+
+    let mut mode1 = Mode::empty();
+    mode1.insert(Mode::S_IRUSR);
+    mode1.insert(Mode::S_IWUSR);
+    fchmod(file.as_raw_fd(), mode1).unwrap();
+
+    let file_stat1 = stat(&filename).unwrap();
+    assert_eq!(file_stat1.st_mode & 0o7777, mode1.bits());
+
+    let mut mode2 = Mode::empty();
+    mode2.insert(Mode::S_IROTH);
+    fchmod(file.as_raw_fd(), mode2).unwrap();
+
+    let file_stat2 = stat(&filename).unwrap();
+    assert_eq!(file_stat2.st_mode & 0o7777, mode2.bits());
+}
+
+#[test]
+fn test_fchmodat() {
+    let tempdir = TempDir::new("nix-test_fchmodat").unwrap();
+    let filename = tempdir.path().join("foo.txt");
+    File::create(&filename).unwrap();
+
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+
+    let mut mode1 = Mode::empty();
+    mode1.insert(Mode::S_IRUSR);
+    mode1.insert(Mode::S_IWUSR);
+    fchmodat(Some(dirfd), &filename, mode1, FchmodatFlags::FollowSymlink).unwrap();
```

Comment:
Here you're supplying an absolute path for `filename`.  I think you should just supply the relative path, "foo.txt".